kotsadm/pkg/kurl/kurl_nodes.go

```diff
@@ -138,20 +143,16 @@ func getNodeMetrics(nodeIP string) (*statsv1alpha1.Summary, error) {
 			return nil, errors.Wrap(err, "get client keypair")
 		}
 
-		client := http.Client{
-			Transport: &http.Transport{
-				TLSClientConfig: &tls.Config{
-					Certificates:       []tls.Certificate{cert},
-					InsecureSkipVerify: true,
-				},
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				Certificates:       []tls.Certificate{cert},
+				InsecureSkipVerify: true,
 			},
 		}
-		r, err = client.Get(fmt.Sprintf("https://%s:10250/stats/summary", nodeIP))
```

Comment:
When this returns an error `r` is nil and only the err in this script gets overwritten. This was causing the `defer r.Body.Close()` call on line 158 below to panic.